odl/solvers/util/callback.py

```diff
@@ -854,6 +855,113 @@ def __repr__(self):
             self.logy)
 
 
+class CallbackPrintHardwareUsage(SolverCallback):
+
+    """Callback for printing memory and CPU usage.
+
+    To use this callback, the user needs to have the package ``psutil``
+    installed.
+    """
+
+    def __init__(self, step=1, print_cpu=True, print_mem=True,
+                 print_swap=True, fmt_cpu='CPU usage (% each core): {}',
+                 fmt_mem='RAM usage: {}', fmt_swap='SWAP usage: {}'):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        step : positive int, optional
+            Number of iterations between output. Default: 1
+        print_cpu : bool, optional
+            If ``True`` then the CPU usage is printed. Default: ``True``
+        print_mem : bool, optional
+            If ``True`` then the RAM memory usage is printed. Default: ``True``
+        print_swap : bool, optional
+            If ``True`` then the SWAP memory usage is printed.
+            Default: ``True``
+        fmt_cpu : string, optional
+            Formating that should be applied. The CPU usage is printed as ::
+
+                print(fmt_cpu.format(cpu))
+
+            where ``cpu`` is a vector with the percentage of current CPU usaged
+            for each core.
+        fmt_mem : string, optional
+            Formating that should be applied. The RAM usage is printed as ::
+
+                print(fmt_mem.format(mem))
+
+            where ``mem`` is the current RAM memory usaged.
+        fmt_swap : string, optional
+            Formating that should be applied. The SWAP usage is printed as ::
+
+                print(fmt_swap.format(swap))
+
+            where ``swap`` is the current SWAP memory usaged.
+
+        Raises
+        ------
+        ImportError
+            If the package ``psutil`` is not installed.
```

Comment:
This is enough imo, no need to spam doc with info in this dependency. In 98% of cases people will get the error and run pip install psutil without even reading the doc.